File: src/main/java/me/crylonz/Localization.java
package me.crylonz;

import java.util.HashMap;
import java.util.Map;

public class Localization {

    public static Map<String, Object> local = new HashMap<>();

    public Localization() {
        local.put("loc_owner", "Owner");
        local.put("loc_loading", "Loading...");
        local.put("loc_not_owner", "This is not your Deadchest !");
        local.put("loc_infinityChest", "Infinity chest");
        local.put("loc_endtimer", "left");
        local.put("loc_reload", "Reload successfull..");
        local.put("loc_noperm", "You need permission");
        local.put("loc_nodc", "You don't have any deadchest");
        local.put("loc_nodcs", "There is currently no deadchest");
        local.put("loc_dclistall", "List of all dead chests");
        local.put("loc_dclistown", "List of your dead chests");
        local.put("loc_doubleDC", "You can't put a chest next to a Deadchest");
        local.put("loc_maxHeight", "You are dead above the maximum height.");
        local.put("loc_noDCG", "No deadchest generated.");
        local.put("loc_givebackInfo", "This player is offline or don't have any active deadchest");
        local.put("loc_dcgbsuccess", "The oldest deadchest content of this player returned to him");
        local.put("loc_gbplayer", "You have retrieved the content of your deadchest");
        local.put("loc_chestPos", "Your deadchest is at");
    }

    public Map<String, Object> get() {
        return local;
    }

    public void set(Map<String, Object> local) {
        Localization.local = local;
    }
}

File: deadchest-core/src/main/kotlin/me/crylonz/deadchest/Localization.kt
package me.crylonz.deadchest

open class Localization {

    private var local: MutableMap<String, Any> = HashMap()

    fun Localization() {
        local["loc_prefix"] = "[DeadChest] "
        local["holo_owner"] = "Deadchest of §6%player%"
        local["holo_timer"] = "§f%hours%§7h §f%min%§7m §f%sec%§7s left"
        local["holo_loading"] = "Loading..."
        local["loc_not_owner"] = "§cThis is not your Deadchest !"
        local["loc_infinityChest"] = "Infinity chest"
        local["loc_endtimer"] = "left"
        local["loc_reload"] = "§6Reload successfull.."
        local["loc_noperm"] = "§cYou need permission"
        local["loc_nodc"] = "§aYou don't have any deadchest"
        local["loc_nodcs"] = "§aThere is currently no deadchest"
        local["loc_dclistall"] = "§aList of all dead chests"
        local["loc_dclistown"] = "§aList of your dead chests"
        local["loc_doubleDC"] = "§cYou can't put a chest next to a Deadchest"
        local["loc_maxHeight"] = "§cYou are dead above the maximum height."
        local["loc_noDCG"] = "§cNo deadchest generated."
        local["loc_givebackInfo"] = "§cThis player is offline or don't have any active deadchest"
        local["loc_dcgbsuccess"] = "§aThe oldest deadchest content of this player returned to him"
        local["loc_gbplayer"] = "§aYou have retrieved the content of your deadchest"
        local["loc_chestPos"] = "§6Your deadchest is at"
        local["loc_noPermsToGet"] = "§cYou are not allowed to open this chest"
    }

    fun set(localToSet: MutableMap<String, Any>) {
        local = localToSet
    }

    fun get(): MutableMap<String, Any> = local

    fun get(key: String): String = local[key].toString()

    fun replacePlayer(localisation: String, playerName: String): String {
        return localisation.replace("%player%", playerName)
    }

    fun replaceTimer(loc: String, hours: Long, min: Long, sec: Long): String {
        var localisation = loc
        localisation = localisation.replace("%hours%", hours.toString())
        localisation = localisation.replace("%min%", min.toString())
        localisation = localisation.replace("%sec%", sec.toString())
        return localisation
    }
}